File: 1871-jump-game-vii/1871-jump-game-vii.cpp
class Solution {
public:
    bool canReach(string s, int minJump, int maxJump) {
        int n = s.length();
        
        vector<bool> dp(n, false);
        dp[0] = true;
        int count = 0;
        
        for(int i = 1; i < n; ++i) {
            if(i > maxJump)
                count -= dp[i - maxJump - 1];
            
            if(i >= minJump)
                count += dp[i - minJump];
            
            if(count > 0 && s[i] == '0')
                dp[i] = true;
        }
        return dp[n - 1];
    }
};

File: 1871-jump-game-vii/1871-jump-game-vii.py
class Solution:
    def canReach(self, s: str, minJump: int, maxJump: int) -> bool:
        n = len(s)
        
        dp = [False] * n
        dp[0] = True
        count = 0
        
        for i in range(1, n):
            if i > maxJump and dp[i - maxJump - 1]:
                count -= 1
            
            if i >= minJump and dp[i - minJump]:
                count += 1
            
            if count > 0 and s[i] == '0':
                dp[i] = True
        
        return dp[n - 1]
                